problem:
  # 预测时间长度为dt*horizon
  name: "cartpole"
  dt: 0.05
  horizon:  20
  verbose: false

# 据说是使用.inf就可以在yaml内完成无穷大的设置
OCP_variables:
  status_frame:
    - name: "q"
      size: 2
      lower_bound:
        - -1.0
        - -.INF
      upper_bound:
        - 1.0
        - .INF
    - name: "dq"
      size: 2
      lower_bound:
        - -.INF
        - -.INF
      upper_bound:
        - .INF
        - .INF
  input_frame:
    - name: "F"
      size: 1
      lower_bound:
        - -100
      upper_bound:
        - 100

solver_settings:
  verbose: true
  gen_code: false
#  如果是recompile则会在运行时重新生成.so文件
  recompile: false
  load_lib: false
#  求解器的方案有很多种： IPOPT, ADMM, CUDA_ADMM, SQPMETHOD, MIXED,对应不同的求解方式
  solve_method: MIXED
